message/validation/validator.go

```diff
@@ -0,0 +1,467 @@
+package validation
+
+// validator.go contains main code for validation and most of the rule checks.
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	specqbft "github.com/bloxapp/ssv-spec/qbft"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"golang.org/x/exp/slices"
+
+	"github.com/bloxapp/ssv/networkconfig"
+	ssvmessage "github.com/bloxapp/ssv/protocol/v2/message"
+	"github.com/bloxapp/ssv/protocol/v2/qbft/roundtimer"
+	"github.com/bloxapp/ssv/protocol/v2/ssv/queue"
+	ssvtypes "github.com/bloxapp/ssv/protocol/v2/types"
+	registrystorage "github.com/bloxapp/ssv/registry/storage"
+)
+
+const (
+	// lateMessageMargin is the duration past a message's TTL in which it is still considered valid.
+	lateMessageMargin = time.Second * 3
+
+	// clockErrorTolerance is the maximum amount of clock error we expect to see between nodes.
+	clockErrorTolerance = time.Millisecond * 50
+
+	maxMessageSize             = 8388608        // TODO: define
+	maxConsensusMsgSize        = maxMessageSize // TODO: define
+	maxPartialSignatureMsgSize = maxMessageSize // TODO: define
+	allowedRoundsInFuture      = 2
```

Comment:
I think SignedPartialSignatureMessage max size is in bytes:
8 (OperatorID) +
96 (Signature) +
8 (PartialSignatureMessages.Type) +
8 (PartialSignatureMessages.Slot) + 
13 * (
    8 (PartialSignatureMessage.Signer) +
    96 (PartialSignatureMessage.PartialSignature) +
    32 (PartialSignatureMessage.SigningRoot)
    )
= 1888

SSVMessage with Data as SignedPartialSignatureMessage would be:
8 + 56 + 1888 = 1952